terraform/deployments/monitoring-test/grafana/main.tf

```diff
@@ -0,0 +1,135 @@
+terraform {
+  backend "s3" {
+    bucket  = "govuk-terraform-test"
+    key     = "projects/grafana.tfstate"
+    region  = "eu-west-1"
+    encrypt = true
+  }
+}
+
+provider "aws" {
+  region = "eu-west-1"
+
+  assume_role {
+    role_arn = var.assume_role_arn
+  }
+}
+
+provider "random" {}
+
+data "aws_region" "current" {}
+
+# TODO pass this ARN in from the govuk deployment (via terraform remote state) instead of using a data source
+data "aws_iam_role" "execution" {
+  name = "fargate_execution_role"
+}
+
+# TODO pass this ARN in from the govuk deployment (via terraform remote state) instead of using a data source
+data "aws_iam_role" "task" {
+  name = "grafana_task_role"
+}
+
+data "terraform_remote_state" "govuk" {
+  backend   = "s3"
+  workspace = terraform.workspace
+  config = {
+    bucket   = "govuk-terraform-test"
+    key      = "projects/govuk.tfstate"
+    region   = data.aws_region.current.name
+    role_arn = var.assume_role_arn
+  }
+}
+
+data "terraform_remote_state" "monitoring" {
+  backend = "s3"
+  config = {
+    bucket   = "govuk-terraform-test"
+    key      = "projects/monitoring.tfstate"
+    region   = "eu-west-1"
+    role_arn = var.assume_role_arn
+  }
+}
+
+data "aws_secretsmanager_secret" "github_client_id" {
+  name = "grafana_github_client_id"
+}
+
+data "aws_secretsmanager_secret" "github_client_secret" {
+  name = "grafana_github_client_secret"
+}
+
+
+resource "random_string" "grafana_password" {
+  length  = 64
+  special = false
+}
+
+resource "aws_secretsmanager_secret" "grafana_password" {
+  name = "grafana_password"
+}
+
+resource "aws_secretsmanager_secret_version" "grafana_password" {
+  secret_id     = aws_secretsmanager_secret.grafana_password.id
+  secret_string = random_string.grafana_password.result
+}
+
+locals {
+  log_group = data.terraform_remote_state.govuk.outputs.log_group
+  mesh_name = data.terraform_remote_state.govuk.outputs.mesh_name
+
+  environment_variables = {
+    GF_SECURITY_ADMIN_USER               = "admin",
+    GF_AUTH_GITHUB_ENABLED               = true,
+    GF_AUTH_GITHUB_SCOPES                = "user:email,read:org",
+    GF_AUTH_GITHUB_AUTH_URL              = "https://github.com/login/oauth/authorize",
+    GF_AUTH_GITHUB_TOKEN_URL             = "https://github.com/login/oauth/access_token",
+    GF_AUTH_GITHUB_API_URL               = "https://api.github.com/user",
+    GF_AUTH_GITHUB_ALLOW_SIGN_UP         = true,
+    GF_AUTH_GITHUB_ALLOWED_ORGANIZATIONS = "alphagov",
```

Comment:
Unnecessary now, but we can also restrict login access to people on GOV.UK.

```
GF_AUTH_GITHUB_TEAM_IDS  = "123" # gov-uk team
```

Reply:
Yeah, it is a future thing to do. Might try to do it today, I have to use curl to get the list as you can't get it via GUI directly.

Reply:
No worries about doing this right now, a TODO would be enough.

Reply:
done

Reply:
done